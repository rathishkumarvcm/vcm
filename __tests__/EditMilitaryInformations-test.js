import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import EditMilitaryInformations from '../App/Screens/EditMilitaryInformation/EditMilitaryInfoComponent';

const navigation = { navigate: jest.fn(), getParam: jest.fn() };
// const props = {};
const getProfileCompositeData = jest.fn();

describe('Edit Military Informations', () => {

  // Application Screnshot
  it('snap shot', () => {
    // const tree = renderer.create(<EditMilitaryInformations {...props} getProfileCompositeData={getProfileCompositeData} navigation={navigation} />).toJSON();
    const tree = renderer.create(<EditMilitaryInformations getProfileCompositeData={getProfileCompositeData} navigation={navigation} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  // Application Render
  test('render', () => {
   // const component = shallow(<EditMilitaryInformations {...props} getProfileCompositeData={getProfileCompositeData} navigation={navigation} />);
    const component = shallow(<EditMilitaryInformations getProfileCompositeData={getProfileCompositeData} navigation={navigation} />);

    component.render();
  });

  test('function test', () => {
   // const component = shallow(<EditMilitaryInformations {...props} getProfileCompositeData={getProfileCompositeData} navigation={navigation} />);
    const component = shallow(<EditMilitaryInformations getProfileCompositeData={getProfileCompositeData} navigation={navigation} />);
    const wrapper = new EditMilitaryInformations();
    component.render();
    wrapper.radioButtonClicked();
    wrapper.dropDownMilitaryOnClick();
    wrapper.dropDownBranchOnClick();
    wrapper.dropDownMarineOnClick();
  });
});