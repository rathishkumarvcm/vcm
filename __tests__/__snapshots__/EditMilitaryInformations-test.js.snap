// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Military Informations snap shot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F7FAFF",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    clickable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 0.12,
        "flexDirection": "row",
        "opacity": 1,
        "paddingLeft": "4%",
        "paddingRight": "4%",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 100.5,
          "opacity": 1,
          "width": "25%",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../App/Images/logo.png",
          }
        }
        style={
          Object {
            "alignItems": "flex-start",
            "justifyContent": "flex-start",
            "width": "100%",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "width": "12%",
        }
      }
    />
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": 24.5,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "height": 40,
            "width": 56,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 40,
              },
              Object {
                "backgroundColor": "transparent",
                "marginHorizontal": 5,
              },
              Object {
                "fontFamily": "EvilIcons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "width": "10%",
        }
      }
    />
    <View
      accessible={true}
      clickable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "height": 30,
            "width": 44,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 30,
              },
              Object {
                "backgroundColor": "transparent",
                "marginHorizontal": 5,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "height": 30,
            "width": 44,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 30,
              },
              Object {
                "backgroundColor": "transparent",
                "marginHorizontal": 5,
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      clickable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          style={
            Object {
              "alignItems": "flex-end",
              "justifyContent": "center",
              "marginEnd": 10,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "height": 40,
                "width": 56,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 40,
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "marginHorizontal": 5,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </RNGestureHandlerButton>
    </View>
  </View>
  <RCTScrollView
    style={
      Object {
        "flex": 0.85,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginTop": 29.5,
            "paddingLeft": "4%",
            "paddingRight": "4%",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#B2B2B2",
              "fontSize": 26.5,
            }
          }
        >
          Settings &gt; Personal info &gt; 
        </Text>
        <Text
          style={
            Object {
              "color": "#B2B2B2",
              "fontSize": 26.5,
              "fontWeight": "bold",
            }
          }
        >
          Manage Military Information
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginTop": 29.5,
            "paddingLeft": "4%",
            "paddingRight": "4%",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#707070",
              "fontSize": 33,
              "width": "70%",
            }
          }
        >
          Manage Military Information
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#B2B2B2",
            "marginLeft": "4%",
            "marginRight": "4%",
            "marginTop": 16.5,
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginTop": 33,
            "paddingLeft": "4%",
            "paddingRight": "4%",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#56565A",
              "fontSize": 29.5,
              "lineHeight": 30,
            }
          }
        >
          Please Provide current military information. We want to ensure that our members receive ourmost competitive offers and rates and are addressed according to correct military protocol
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginTop": 33,
            "paddingLeft": "4%",
            "paddingRight": "4%",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#333333DE",
              "fontSize": 29.5,
              "fontWeight": "bold",
            }
          }
        >
          Are you currently serving, or have you ever served in the US Military?
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginTop": 33,
            "paddingLeft": "4%",
            "paddingRight": "4%",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginLeft": "2%",
              "marginTop": 3.5,
              "width": "40%",
            }
          }
        >
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": "2%",
                "marginTop": "4%",
                "opacity": 1,
                "width": "92%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#707070",
                  "borderRadius": 26.5,
                  "borderWidth": 3.5,
                  "height": 52.5,
                  "justifyContent": "center",
                  "width": 52.5,
                }
              }
            />
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "column",
                    "height": 82,
                    "marginLeft": "5%",
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#333333DE",
                    "fontSize": 26.5,
                  }
                }
              >
                Yes
              </Text>
              <Text
                style={
                  Object {
                    "alignItems": "flex-start",
                    "color": "#333333DE",
                    "fontSize": 19.5,
                    "marginTop": "5%",
                  }
                }
              />
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": "2%",
                "marginTop": "4%",
                "opacity": 1,
                "width": "92%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#707070",
                  "borderRadius": 26.5,
                  "borderWidth": 3.5,
                  "height": 52.5,
                  "justifyContent": "center",
                  "width": 52.5,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#707070",
                    "borderRadius": 10,
                    "height": 19.5,
                    "width": 19.5,
                  }
                }
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "column",
                    "height": 82,
                    "marginLeft": "5%",
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#333333DE",
                    "fontSize": 26.5,
                  }
                }
              >
                No
              </Text>
              <Text
                style={
                  Object {
                    "alignItems": "flex-start",
                    "color": "#333333DE",
                    "fontSize": 19.5,
                    "marginTop": "5%",
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginTop": 33,
            "paddingLeft": "4%",
            "paddingRight": "4%",
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#56565A",
              "borderRadius": 1,
              "borderWidth": 1,
              "height": 82,
              "justifyContent": "center",
              "marginLeft": "4%",
              "marginRight": "4%",
              "marginTop": 19.5,
              "opacity": 1,
              "width": "92%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#56565A",
                  "fontSize": 26.5,
                },
              ]
            }
          >
            Cancel
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "marginTop": 33,
            "paddingLeft": "4%",
            "paddingRight": "4%",
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#56565A",
              "borderColor": "#56565A",
              "borderRadius": 1,
              "borderWidth": 1,
              "height": 82,
              "justifyContent": "center",
              "marginLeft": "4%",
              "marginRight": "4%",
              "marginTop": 19.5,
              "opacity": 1,
              "width": "92%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 26.5,
                },
              ]
            }
          >
            Save
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "marginTop": 39.5,
            "padding": 24.5,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#56565A",
              "fontSize": 29.5,
              "marginBottom": 8,
            }
          }
        >
          Investing involves risk including loss of principal.
        </Text>
        <Text
          style={
            Object {
              "color": "#56565A",
              "fontSize": 26.5,
              "justifyContent": "center",
            }
          }
        >
          Victory Mutual Funds and USAA Mutual Funds are distributed by Victory Capital Advisers, Inc. (VCA). VictoryShares ETFs and VictoryShares USAA ETFs are distributed by Foreside Fund Services, LLC (Foreside). VCA and Foreside are members of FINRA and SIPC. Victory Capital Management Inc. (VCM) is the investment adviser to the Victory Mutual Funds, USAA Mutual Funds, VictoryShares ETFs and VictoryShares USAA ETFs. VCA and VCM are not affiliated with Foreside. USAA is not affiliated with Foreside, VCM, or VCA. USAA and the USAA logos are registered trademarks and the USAA Mutual Funds and USAA Investments logos are trademarks of United Services Automobile Association and are being used by Victory Capital and its affiliates under license. Victory Capital means Victory Capital Management Inc., the investment manager of the USAA 529 College Savings Plan (Plan). The Plan is distributed by Victory Capital Advisers, Inc., a broker dealer registered with FINRA and an affiliate of Victory Capital. Victory Capital and its affiliates are not affiliated with United Services Automobile Association or its affiliates. USAA and the USAA logo are registered trademarks and the USAA 529 College Savings Plan logo is a trademark of United Services Automobile Association and are being used by Victory Capital and its affiliates under license.
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "flexDirection": "column",
            "height": 164.5,
            "paddingBottom": "5%",
            "paddingLeft": "5%",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../App/Images/logo.png",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flexDirection": "row",
            "height": 82,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5D83AE",
              "fontSize": 26.5,
              "paddingLeft": "4%",
              "width": "50%",
            }
          }
        >
          Privacy Policy
        </Text>
        <Text
          style={
            Object {
              "color": "#5D83AE",
              "fontSize": 26.5,
              "paddingLeft": "4%",
              "width": "50%",
            }
          }
        >
          Fund Documents
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flexDirection": "row",
            "height": 82,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#5D83AE",
              "fontSize": 26.5,
              "paddingLeft": "4%",
              "width": "50%",
            }
          }
        >
          User Agreement
        </Text>
        <Text
          style={
            Object {
              "color": "#5D83AE",
              "fontSize": 26.5,
              "paddingLeft": "4%",
              "width": "50%",
            }
          }
        >
          Support
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#56565A",
            "height": 82,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          Copyright Victory Capital Management Inc. ©2020
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
